L3
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
L2
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
L1
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
L2
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
L2
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
L2
0      False
1      False
2      False
3      False
4      False
       ...
666    False
667    False
668    False
669    False
670    False
Name: ID, Length: 671, dtype: bool
Traceback (most recent call last):
  File "train.py", line 127, in <module>
    train_5_fold(dataset, batch_size=16, epochs=50, learning_rate=0.0001, optimizer_type='AdamW', scheduler_type='CosineAnnealingLR', device='cuda')
  File "train.py", line 49, in train_5_fold
    train_loss = train_one_epoch(model, train_loader, optimizer, criterion, device)
  File "train.py", line 85, in train_one_epoch
    for images, labels, _ in train_loader:
  File "C:\Users\hyunoh\anaconda3\envs\spine\lib\site-packages\torch\utils\data\dataloader.py", line 630, in __next__
    data = self._next_data()
  File "C:\Users\hyunoh\anaconda3\envs\spine\lib\site-packages\torch\utils\data\dataloader.py", line 673, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "C:\Users\hyunoh\anaconda3\envs\spine\lib\site-packages\torch\utils\data\_utils\fetch.py", line 52, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "C:\Users\hyunoh\anaconda3\envs\spine\lib\site-packages\torch\utils\data\_utils\fetch.py", line 52, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "C:\Users\hyunoh\Documents\Codes\BMD_code\regression_function_classes.py", line 207, in __getitem__
    bmd = self.get_bmd(os.path.basename(img_path), ID)
  File "C:\Users\hyunoh\Documents\Codes\BMD_code\regression_function_classes.py", line 214, in get_bmd
    bmd = self.bmd_data[self.bmd_data['ID'] == ID][vert_name].values[0]
IndexError: index 0 is out of bounds for axis 0 with size 0
